### DISCUSSION TITLE
Header based routing for pubsub workers

### DISCUSSION THREAD

**Discussion by AlonGluz** [linkerd/linkerd2#10814] [Q&A]
Hey,

It was great meeting you at Kubecon. 
We are using Linkerd with Flagger to deploy new versions of services. We have an issue that during a canary rollout, there is some sort of a race condition over the messages storage by a topic, causing some level of uncertainty when consumers read the messages. 
I was wondering if I can somehow utilize the most awesome feature in the world, header-based routing to do canary rollouts over Pubsub.

**Comment by adleong** [comment]
Hi @AlonGluz!

I think the answer depends on the details of what you're using for pubsub.  Linkerd can only do header-based routing on HTTP, which means that if your pubsub system uses HTTP, then header-based routing should work for you.  Otherwise, header-based routing would not apply.
